packages/Breadcrumbs/src/Breadcrumbs.js

```diff
@@ -11,14 +12,15 @@ import * as sc from "./Breadcrumbs.styles";
 function Breadcrumbs(props) {
   const { children, isDark, isAutoCollapsed, ...moreProps } = props;
   const I18n = useI18n();
-  const childrenCount = React.Children.count(children);
-  const shouldShowExpandButton = isAutoCollapsed && React.Children.count(children) > MAXIMUM_NUM_OF_LEVEL;
-  const hasOnlyOneChild = childrenCount === 1;
+
   const [isCollapsed, setIsCollapsed] = React.useState(shouldShowExpandButton);
+  const linkChildren = [extractChildren(children, ["Breadcrumbs.Link"])["Breadcrumbs.Link"]].flat();
+  const hasOnlyOneChild = linkChildren.length === 1;
+  const shouldShowExpandButton = isAutoCollapsed && linkChildren.length > MAXIMUM_NUM_OF_LEVEL;
```

Comment:
```js
function Breadcrumbs({ initialmaxNumLevels }: { initialmaxNumLevels = Infinity }) {
  const [maxNumLevels, setMaxNumLevels] = React.useState(initialmaxNumLevels);
  const isCollapsed = React.Children.count(children) > maxNumLevels;
  
  function handleExpand() {
    setMaxNumLevels(Infinity);
  }
}
```

Reply:
```js
const links = [<Breadcrumbs.Link />, <Breadcrumbs.Link />];

React.useEffect(() => {
  fetchSomeLinks().then(data => links.push(data.map(link => <Breadcrumbs.Link link={link}/>)))
}, [])

<Breadcrumbs initialmaxNumLevels={2}> 
  {links}
</Breadcrumbs>
```

after the fetch is done, the `links.length` will become `2 + N` so
`const isCollapsed = React.Children.count(children) > maxNumLevels;` will become true

after we click `handleExpand`, we set `setMaxNumLevels(Infinity);` -> `const isCollapsed = React.Children.count(children) > maxNumLevels;` becomes false

Reply:
I like this API better, it's more flexible and potentially less error prone.  But I'm not sure it's worth breaking the API for it.  

Instead I've mocked out an async scenario in Storybook, and can verify it still behaves correctly without the somewhat awkward `useLayoutEffect()`.

Reply:
If you click on the `...` before the data loaded in this example http://storybooks.highbond-s3.com/paprika/UXD-1717--filter-breadcrumb-children/?path=/story/navigation-breadcrumbs-examples--dynamic
You can see the breadcrumbs remain expanded after loading
That was what I want to avoid in the initial version, but I'm ok with the current behavior as long as we're aware of it 😄

Reply:
I double checked with @carolchaan:

> I think what you and Andriy have is fine cause it's an extra step for them to click the expand again and there's no way to undo an expand anyway